modules/rostests/apitests/gdi32/GetDIBits.c

```diff
@@ -19,8 +19,22 @@ Test_GetDIBits_xBpp(
     ULONG cjSizeImage, cColors;
     HDC hdc;
 
-    hdc = GetDC(0);
+    hdc = GetDC(NULL);
+    ok(hdc != NULL, "GetDC failed\n");
+    if (hdc == NULL)
+    {
+        skip("No hdc\n");
+        return;
+    }
+
     hbmp = CreateBitmap(3, 3, 1, cBitsPixel, NULL);
+    ok(hbmp != NULL, "CreateBitmap failed\n");
+    if (hbmp == NULL)
+    {
+        skip("No hbmp\n");
+        ok(ReleaseDC(NULL, hdc) == 1, "ReleaseDC failed\n");
```

Comment:
ReleaseDC (and DeleteObject, see below) are functions, while ok() is a macro. You should never call a function within a parameter of a macro, because nothing guarantees that the macro won't evaluate more than one time its parameters, and thus, here, calling the function more than once.
You should therefore call the function and retrieve its result, and then use the ok() macro using only the result.

Separately from this general remark, I'm not sure it makes real sense to check whether the release/delete of DCs (or other **valod** objects) has succeeded, since releasing/deleting a valid object must succeed (as well, a specific test for these Create/Select/Release/Delete functions where these checks are done, should already exist, so we should know whether the functions in question behave correctly, already).

Reply:
I don't agree, unfortunately ;)

While the macro thing may be generally a good rule, `ok` is well-defined for this purpose. There are hundreds of places where it's called this way. The only reason this isn't good practice is that in case of failure you can't easily see the actual result for debugging.

As for testing unnecessary calls... this can be annoying if the "unrelated" feature is unreliable, as it causes failures in unrelated tests; but it can also find real bugs and broken assumptions in the test. Since there's no clear answer, my advice would simply be, as so often: don't change it in existing code.

Reply:
I disagree on both remarks:
1. What Thomas answered.
I might even argue that unsafe macros are bad...
2. Checking cleanup functions is not about testing them, it's about (fully) checking how the current test/run behaves.
Thus, I don't understand that last advice: I think tests should explicitly check and report, as much as possible! (Especially while developing an alpha-stage OS...)

Reply:
Well then, why not testing the printf and the free() calls as well in every unrelated tests? 🤣 🤣

Reply:
??why not??

Sure, go ahead!
And let me know if you find other `GDI32` related functions...